volume-providers/vpc/provider/wait_for_attach_volume.go

```diff
@@ -27,7 +28,26 @@ func (vpcs *VPCSession) WaitForAttachVolume(volumeAttachmentTemplate provider.Vo
 	if err != nil {
 		return nil, err
 	}
-	maxTimeout, maxRetryAttempt, retryGapDuration := vpcs.Config.GetTimeOutParameters()
+	err = FlexyRetry(vpcs.Logger, func() (interface{}, error) {
+		currentVolAttachment, errAPI := vpcs.GetVolumeAttachment(volumeAttachmentTemplate)
+		return currentVolAttachment, errAPI
+	}, func(intf interface{}, err *models.Error) bool {
+		if err != nil {
+			for _, errorItem := range err.Errors {
+				skipStatus, ok := skipErrorCodes[string(errorItem.Code)]
```

Comment:
Can you move this skipErrorCode logic in common functions  so that it can be called from multiple callers?

Reply:
done